### PR TITLE
Allow adding health check callback to Dapr App run in gRPC

### PR DESCRIPTION
**Pull request by evhen14**
# Description

Add health check support to the SDK for gRPC following the example of the go-sdk. The dapr docs state that the `/dapr.proto.runtime.v1.AppCallbackHealthCheck/HealthCheck` endpoint is called, but this SDK doesn't support specifying that method implementation. 

The suggested way to use it would be `app.register_health_check(health_check_func)`

**Use case 1**: gRPC endpoints depend on other components to function properly. The health checks callback can be used to validate the liveness of the other components

**Use case 2**: Run gRPC app next to a web app in the same container in a separate thread. The entire app is healthy only when both dapr app and web app are healthy. I expect that dapr thread could die or stop being responsive, hence the way to add health checks.

## Issue reference

It's the outcome of the issue #722 I raised earlier where I see a gap for adding meaningful health checks

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: ext/dapr-ext-grpc/dapr/ext/grpc/_health_servecier.py
Comment (passuied) [inline ext/dapr-ext-grpc/dapr/ext/grpc/_health_servecier.py:1]: typo: should be `_health_servicier.py`

**Comment by berndverst** [comment]
Please run `tox -e ruff` for our autoformatter of source files.

**Review by berndverst — COMMENTED** [review]
Great contribution - and LGTM to me. Please run `tox -e ruff` to apply our autoformatter for code and commit the changes.

I'll approve and merge once that is done.

**Comment by yevgen-el8** [comment]
@berndverst `tox -e ruff` was applied. Thx for your feedback!

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
```
ext/dapr-ext-grpc/dapr/ext/grpc/_health_servicer.py:17: error: Function "cb" could always be true in boolean context  [truthy-function]
ext/dapr-ext-grpc/dapr/ext/grpc/_health_servicer.py:22: error: Attribute "_health_check_cb" already defined on line 19  [no-redef]
ext/dapr-ext-grpc/dapr/ext/grpc/_health_servicer.py:22: error: Incompatible types in assignment (expression has type "None", variable has type "Callable[[], None]")  [assignment]
```

**Review by berndverst — APPROVED** [review]

**Comment by marcduiker** [comment]
@holopin-bot @evhen14 Thank you!

**Pull request merged**